### PR TITLE
[utils] `traverse_obj`: Move `is_user_input` into output template

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Preparations for `traverse_obj` rewrite in combination with output template adjustments.
Both private parameters should be moved out of the function and into `outtmpl` instead.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
> but it would be nice if we could separate concerns and not have a single, massive traversal + outtmpl PR

We'll be able to split it up after we have the full picture.

**Comment by pukkandan** [comment]
Raise a deprecation_warning if is_user_input is being used. We can remove the param completely after a while.

File: yt_dlp/YoutubeDL.py [hunk 1]
```diff
@@ -1200,6 +1200,15 @@ def prepare_outtmpl(self, outtmpl, info_dict, sanitize=False):
                 (?:\|(?P<default>.*?))?
             )$''')
 
+        def _from_user_input(field):
+            if field == ':':
+                return ...
+            elif ':' in field:
+                return slice(*map(int_or_none, field.split(':')))
+            elif int_or_none(field) is not None:
+                return int(field)
```
[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/YoutubeDL.py:1209]: imo, if you have to use `()` around `:=`, it probably shouldn't be used. But use whichever you prefer.

[the remark this answers is not in the document]
Comment (Grub4K) [reply yt_dlp/YoutubeDL.py:1209]: Ill leave it then

**Comment by pukkandan** [comment]
Even so, no harm in leaving it for a bit just in case.

**Review by bashonly — APPROVED** [review]

**Pull request merged**